csharp/gen_csharp_binding.ml

```diff
@@ -115,16 +114,14 @@ let joined sep f l =
   String.concat sep
     (List.filter (fun x -> String.compare x "" != 0) r)
 
+let escape_xml s = s |>
+  Astring.String.cuts ~sep:"<" ~empty:true |>
+  String.concat "&lt;" |>
+  Astring.String.cuts ~sep:">" ~empty:true |>
+  String.concat "&gt;"
```

Comment:
This is very declarative and `Astring` guarantees the invariants with `~empty:true`, I like it. A faster the alternative could be

```ocaml
let buf = Buffer.create (String.length s) in
List.iter (function
| '<' -> Buffer.add_string buf "&lt;"
| '>' -> Buffer.add_string buf "&gt;"
| c   -> Buffer.add_char buf c) s;
s |> Buffer.to_bytes |> Bytes.to_string
```

I like your version more, so if speed is not a concern I would not change it.

Reply:
I expect the speed difference to be minimal since we don't handle big strings - overall the build is very fast and the biggest delay comes from io actions in the Makefile rather than the jbuilder stuff (it's in my todo list to investigate doing those too via jbuilder).